### PR TITLE
Refactoring Unit Tests

### PR DESCRIPTION
**Pull request by rstackhouse**
* Switching to .NET Core 3.1
* Simplifying test
* Implementing additional tests
* Refactoring tests

### CODE REVIEW

File: Automation/LettersDispatcherTest.cs [hunk 1]
```diff
@@ -28,102 +28,145 @@ public LettersDispatcherTest()
...
+
+            // Act
+            this.lettersDispatcher.Dispatch(letters);
+
+            // Assert
+            receiver.ReceivedLetters.Should().HaveCount(1); // Input drives output. No need for exact equality test.
+        }
+
+        [Fact]
+        public void Should_SendLetterToReceivers_When_ThereIsNoMessage()
```
Comment (LeanHer94) [inline Automation/LettersDispatcherTest.cs:135]: change to Should_NotSendLetterToReceivers_When_ThereIsNoMessage

Comment (leandro-hl) [inline Automation/LettersDispatcherTest.cs]: change to Should_NotSendLetterToReceivers_When_ThereIsNoMessage

File: Automation/LettersDispatcherTest.cs [hunk 2]
```diff
@@ -28,102 +28,145 @@ public LettersDispatcherTest()
...
+
+            // Act
+            this.lettersDispatcher.Dispatch(letters);
+
+            // Assert
+            receiver.ReceivedLetters.Should().HaveCount(0);
+        }
+
+        [Fact]
+        public void Should_SendLetterToReceivers_When_ThereIsNoSubject()
```
Comment (leandro-hl) [inline Automation/LettersDispatcherTest.cs]: change to Should_NotSendLetterToReceivers_When_ThereIsNoSubject

Comment (rstackhouse) [reply Automation/LettersDispatcherTest.cs]: Naming fixed.

File: Automation/LettersDispatcherTest.cs (hunk 1 above)
Comment (rstackhouse) [reply Automation/LettersDispatcherTest.cs]: Naming fixed.

**Review by leandro-hl — APPROVED** [review]

**Pull request merged**